dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/controller/RequestController.kt

```diff
@@ -28,15 +31,47 @@ class RequestController(
     }
 
     override fun getDataRequestsForUser(): ResponseEntity<List<StoredDataRequest>> {
-        return ResponseEntity.ok(
-            dataRequestManager.getDataRequestsForUser(),
-        )
+        return ResponseEntity.ok(dataRequestManager.getDataRequestsForUser())
     }
 
     override fun getAggregatedDataRequests(
         identifierValue: String?,
         dataTypes: Set<DataTypeEnum>?,
-    ): List<AggregatedDataRequest> {
-        return dataRequestManager.getAggregatedDataRequests(identifierValue, dataTypes)
+        reportingPeriod: String?,
+    ): ResponseEntity<List<AggregatedDataRequest>> {
+        return ResponseEntity.ok(
+            dataRequestManager.getAggregatedDataRequests(identifierValue, dataTypes, reportingPeriod),
+        )
+    }
+
+    override fun postSingleDataRequest(singleDataRequest: SingleDataRequest): ResponseEntity<List<StoredDataRequest>> {
+        return ResponseEntity.ok(dataRequestManager.processSingleDataRequest(singleDataRequest))
+    }
+
+    override fun getDataRequestById(dataRequestId: UUID): ResponseEntity<StoredDataRequest> {
+        return ResponseEntity.ok(dataRequestManager.getDataRequestById(dataRequestId.toString()))
+    }
+
+    override fun patchDataRequest(
+        dataRequestId: UUID,
+        requestStatus: RequestStatus,
+    ): ResponseEntity<StoredDataRequest> {
+        return ResponseEntity.ok(dataRequestManager.patchDataRequest(dataRequestId.toString(), requestStatus))
+    }
+
+    override fun getDataRequests(
+        dataType: DataTypeEnum?,
+        userId: String?,
+        requestStatus: RequestStatus?,
+        reportingPeriod: String?,
+        dataRequestCompanyIdentifierValue: String?
+    ): ResponseEntity<List<StoredDataRequest>> {
+        return ResponseEntity.ok(dataRequestManager.getDataRequests(
+            dataType,
+            userId,
+            requestStatus,
+            reportingPeriod,
+            dataRequestCompanyIdentifierValue
+        ))
```

Comment:
Would adding correlationId to the requests here, like we have for the data storage have merit?

Reply:
@emanueldfine Stephan and I talked about it briefly - do you have a strong opinion on that? Otherwise, we would just leave everything as is due to the probably bad cost-performance ratio (argument: except for querying companies from the backend, there is no interaction between different microservices)

Reply:
ok, i will close this, i dont think it has a lot of merit right now